### PR TITLE
Add tests for %TypedArray%.prototype.findIndex

### PR DESCRIPTION
**Pull request by leobalter**

### CODE REVIEW

File: test/built-ins/TypedArray/prototype/findIndex/predicate-call-this-non-strict.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  sample.findIndex(function() {
+    result = this;
+  });
+
+  assert.sameValue(result, T, "without thisArg, predicate this is the global");
+
+  var o = {};
+  sample.findIndex(function() {
+    result = this;
+  }, o);
```
Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/findIndex/predicate-call-this-non-strict.js]: I just thought of another assertion that might be good here (and in the tests for `find`): specifying `undefined` for `thisArg`. In non-strict mode, the "this" value should *still* be the global object. Do you think that's worth testing?

File: test/built-ins/TypedArray/prototype/findIndex/predicate-is-not-callable-throws.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    sample.findIndex(null);
+  }, "null");
+
+  assert.throws(TypeError, function() {
+    sample.findIndex(undefined);
+  }, "undefined");
+
+  assert.throws(TypeError, function() {
+    sample.findIndex(false);
+  }, "true");
```
Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/findIndex/predicate-is-not-callable-throws.js]: "true" -> "false"

File: test/built-ins/TypedArray/prototype/findIndex/return-negative-one-if-predicate-returns-false-value.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  assert.sameValue(result, -1, "ToBoolean(string)");
+
+  result = sample.findIndex(function(val) { return undefined; });
+  assert.sameValue(result, -1, "ToBoolean(undefined)");
+
+  result = sample.findIndex(function(val) { return null; });
+  assert.sameValue(result, -1, "ToBoolean(null)");
+
+  result = sample.findIndex(function(val) { return 0; });
+  assert.sameValue(result, -1, "ToBoolean(0)");
```
Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/findIndex/return-negative-one-if-predicate-returns-false-value.js]: Mind adding an assertion for -0 here, too?

**Comment by jugglinmike** [comment]
Just a few minor requests for this one, Leo, then it should be good to go.

File: test/built-ins/TypedArray/prototype/findIndex/predicate-call-this-non-strict.js (hunk 1 above)
Comment (leobalter) [inline test/built-ins/TypedArray/prototype/findIndex/predicate-call-this-non-strict.js]: +1. I could tell the non existent argument is handled as undefined, but it's precise adding a specific test for it.

File: test/built-ins/TypedArray/prototype/findIndex/predicate-is-not-callable-throws.js (hunk 2 above)
Comment (leobalter) [inline test/built-ins/TypedArray/prototype/findIndex/predicate-is-not-callable-throws.js]: oops. at least the assertion is valid.

Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/findIndex/predicate-is-not-callable-throws.js]: Yup!

**Comment by leobalter** [comment]
Thanks! fixes at 496f0ac

File: test/built-ins/TypedArray/prototype/findIndex/predicate-call-this-non-strict.js [hunk 4]
```diff
@@ -39,14 +39,16 @@ testWithTypedArrayConstructors(function(TA) {
...
+
+  result = null;  
   sample.findIndex(function() {
     result = this;
   }, undefined);
 
   assert.sameValue(result, T, "predicate this is the global when thisArg is undefined");
 
   var o = {};
+  result = null;
```
Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/findIndex/predicate-call-this-non-strict.js]: Good catch

**Comment by jugglinmike** [comment]
Looks good to me!

**Comment by jugglinmike** [comment]
Rebased, squashed, and merged to `master` at commit bc0a40c51ce9dad6a0fe4f5f8967559d7b707d8b. Thanks, Leo!

**Pull request closed without merging**